### PR TITLE
Use command -v to detect git

### LINKED ISSUE #81287: Git: "git not found" error, macOS, when xcodebuild license needs renewing
It appears that on macOS if xcodebuild license needs renewing VSCode throws a "git not found" error.

Screen Shot 2019-09-22 at 10 23 52 AM

```
Lukes-iMac-2:skatex lukepighetti$ git

Agreeing to the Xcode/iOS license requires admin privileges, please run “sudo xcodebuild -license” and then retry this command.
```

Aaaaand now it works

### PR DESCRIPTION
**Pull request by radar**
Version details:

```
Version: 1.39.1
Commit: 88f15d17dca836346e787762685a40bb5cce75a8
Date: 2019-10-10T23:35:11.314Z
Electron: 4.2.10
Chrome: 69.0.3497.128
Node.js: 10.11.0
V8: 6.9.427.31-electron.0
OS: Darwin x64 18.7.0
```

## What I expect to happen

VS Code's Git extension finds + uses Git

## What actually happens

VS Code says it cannot find Git:

image

## Steps to reproduce

TL;DR: VS Code should use `command -v` to detect commands, not `which`.

This is a bit tricky, so bear with me:

There is a node package called which:

https://www.npmjs.com/package/which

If this package is installed through Node, _and_ a user is using asdf, this which-from-node can take precedence in $PATH. This will cause issues if there is no default Node version specified for asdf. An error like this can appear:

```
asdf: No version set for command which
you might want to add one of the following in your .tool-versions file:

nodejs 8.14.0
```

(I suspect this is only an issue on 'zsh' terminals -- I am unable to reproduce it in Bash)

To work around this which-conflict, we should use the built-in 'command' command. This comes standard in every shell program and is a viable alternative to 'which'.

When my shell works correctly, here's the output of `which -a which`:

```
which: shell built-in command
/Users/ryanbigg/.asdf/shims/which
/usr/bin/which
```

However, when it is misbehaving, I _suspect_ the order is like this:

```
/Users/ryanbigg/.asdf/shims/which
which: shell built-in command
/usr/bin/which
```

And that's why it's trying to use the node `which`. When no node version is configured for `asdf`, then it fails to run that node-which, which then causes this extension to claim that it can't find git.

This PR (likely) fixes #81287 -- although I'm not 100% sure of that.

### CODE REVIEW

**Comment by JoshTeperman** [comment]
👀

**Comment by joaomoreno** [comment]
So... what happens if the user has a `command` command ahead in their PATH variable, just like `which`?

**Comment by joaomoreno** [comment]
Closing.

**Comment by ryenus** [comment]
> So... what happens if the user has a `command` command ahead in their PATH variable, just like `which`?

I believe that wouldn't be a problem because the shell builtin `command` should take precedence over things from `$PATH`.

From bash manual: [Shell Builtin Commands][1]
> When the name of a builtin command is used as the first word of a simple command (see Simple Commands), the shell executes the command directly, without invoking another program.

[1]: https://www.gnu.org/software/bash/manual/html_node/Shell-Builtin-Commands.html#Shell-Builtin-Commands

**Comment by joaomoreno** [comment]
Isn't `which` a builtin command?

**Comment by ryenus** [comment]
> Isn't which a builtin command, as was mentioned above?

In *csh* `which` is indeed a builtin, but not in **sh/bash**, @joaomoreno

**Comment by joaomoreno** [comment]
Got it. Keeping my opinion: this is not worth the trouble.
